<?xml version="1.0"?>

<launch>
    <!-- <arg name="camera" default="camera" /> -->

    <!-- start sensor-->
    <!-- <include file="$(find openni2_launch)/launch/openni2.launch">
        <arg name="camera" default="$(arg camera)"/>
    </include> -->

    <!-- run pointcloud_to_laserscan node -->
    <!-- <node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan">

        <remap from="cloud_in" to="lidar3D"/>
        <remap from="scan" to="scan"/>
        <rosparam>
            target_frame: lidar3D # Leave disabled to output scan in pointcloud frame
            transform_tolerance: 0.01
            min_height: -2.5
            max_height: 0.0

            angle_min: -1.6 #-3.141592 # -M_PI/2
            angle_max: 1.6 #3.14/6 #3.141592 # M_PI/2
            angle_increment: 0.0087 # M_PI/360.0
            scan_time: 0.0666667 # 15Hz
            range_min: 0.1
            range_max: 200
            use_inf: true
            inf_epsilon: 1.0

            # Concurrency level, affects number of pointclouds queued for processing and number of threads used
            # 0 : Detect number of cores
            # 1 : Single threaded
            # 2->inf : Parallelism level
            concurrency_level: 1
        </rosparam>

    </node> -->
    
    <node pkg="tf2_ros" type="static_transform_publisher" name="map_to_odom" args="1.5 0.2 1.25 0 0 0 1 map odom" />
    <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="odom_to_base_link" args="1.5 0 1.25 0 0 0 1 odom base_link" /> -->
    <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="base_link_to_scan" args="1.5 0 1.25 0 0 0 1 base_link scan" /> -->
    <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="base_link_to_lidar" args="1.5 0 1.25 0 0 0 1 base_link lidar3D" /> -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="base_link_to_imu" args="0 0 0 0 0 0 1 base_link imu" />
    <node pkg="tf2_ros" type="static_transform_publisher" name="base_link_to_gps" args="0 0 0 0 0 0 1 base_link gps" />

    <node pkg="wego" type="planner.py" name="planner" args="naver_sim.txt 314494.0581980597 4160624.929391573" output="screen" />
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find wego)/rviz/planner.rviz" />

</launch>
